src/common/components/CtaCard/CtaCard.tsx

```diff
@@ -1,27 +1,42 @@
+import {
+  type ReactNode,
+  type ReactElement,
+  isValidElement,
+  cloneElement,
+} from "react";
+
 import {
   Button,
   type ButtonVariants,
   type ButtonLinkOrAnchorProps,
 } from "@/common/components/Button";
-import { EditIcon, PlusIcon } from "@/common/components/CustomIcon";
 
 import { type CtaCardVariants, ctaCardTheme } from "./CtaCard.theme";
 
 export type CtaCardProps = CtaCardVariants &
   ButtonVariants &
   ButtonLinkOrAnchorProps & {
     ctaText: string;
+    leftIcon?: ReactNode;
+    rightIcon?: ReactNode;
   };
 
 export const CtaCard = ({ ctaText, ...props }: CtaCardProps) => {
   const { button, ctaWrapper, cta, icon } = ctaCardTheme();
+  const renderIcon = (iconElement: ReactNode) => {
+    if (isValidElement(iconElement)) {
+      return cloneElement(iconElement as ReactElement, { className: icon() });
+    }
+    return null;
+  };

```

Comment:
question: why do we need this check for `isValidElement`?

Reply:
if it was like this instead
![image](https://github.com/user-attachments/assets/90a4ede1-ca74-4165-9aaf-deeae04a52f7)

and i passed in this (which is a valid ReactNode) as a prop
![image](https://github.com/user-attachments/assets/a6513b11-c8e3-4ff7-9fdd-4ab5cc495652)

then there'd be an error cuz 123 cannot be cloned

but the main point about the renderIcon function is to render nothing if the leftIcon/rightIcon prop isnt passed, and render the icon properly if the icon is passed. I guess as long as we make sure we pass in the Icon components properly then we can just leave it as if (iconElement)?

Reply:
ah, I misphrased my question

what i meant was: why do we not just pass the `leftIcon` and `rightIcon` directly to the Button component since it already also has the same logic?

https://github.com/AfterClass-io/afterclass.io-v2/blob/main/src/common/components/Button/Button.tsx#L140-L159

Reply:
ahh ok, i didnt know that the Button component has that logic already haha

Reply:
I'm actually not sure how to just use the Button component directly inside the CtaCard component, seems like if we were to do that, we have to make quite a few changes, an alternative that I can think of (that I feel isn't as good as what is currently in the PR) is:

1. adding the `const { icon } = ctaCardTheme();` for any component that wants to use the CtaCard component, and passing it in as a prop for the icon `iconLeft={<PlusIcon className={icon()} />}` but this doesn't seem too good
2. the span tag within the button has to be styled separately as well (because the `asChild` prop has to be removed for the left/right icon prop to render properly) so there probably has to be a different prop passed into the Button component that will be applied to the span tag inside the button?

Not sure if you know of any other better way to do this?

Reply:
a bummer that we have to jump through those hoops to now use cloneElement here, seems like this render function is still necessary to inject the class into the component

let's go with your original changes!